### ISSUE TITLE
250814 Deep Learning_문채린

### ISSUE THREAD

**Issue by CHAERINMOON** [AIBootcamp14/wiki-rider#29]
## 📌 키워드 및 요약 정리

🔹 딥러닝(Deep Learning)

      - 기본적으로 인공신경망을 깊게 쌓은 구조
      - 여러 가지 아키텍처가 존재

🔹 CNN (Convolutional Neural Network)

      -이미지/영상/공간 데이터에 특화된 딥러닝 모델
      -합성곱(Convolution)과 풀링(Pooling)을 통해 특징을 자동 추출

🔹 RNN (Recurrent Neural Network)

      -순차적 데이터(시계열, 텍스트, 음성)에 특화된 딥러닝 모델
      -순환 구조로 과거 정보를 기억하면서 현재 계산에 반영

### 1️⃣ 딥러닝(Deep Learning)
          
       - 인공신경망을 깊게 쌓아 올린 모델
       - 데이터에서 특징(feature)을 자동으로 학습하여 복잡한 패턴을 인식
       - 전통적인 머신러닝과 차이점:
       - 머신러닝: 사람이 직접 특징을 추출(feature engineering) → 모델 학습
       - 딥러닝: 모델이 스스로 특징을 추출하여 학습

### 2️⃣ 신경망 기본 개념
           
       - 뉴런(Neuron): 입력 → 가중치(Weight) → 합 → 활성화 함수(Activation) → 출력
       - 층(Layer)
       - 입력층(Input Layer): 데이터 입력
       - 은닉층(Hidden Layer): 데이터를 변환하고 특징을 학습
       - 출력층(Output Layer): 최종 결과(분류/예측)

### (1) CNN (Convolutional Neural Network)
           
       - 합성곱 신경망
       - 이미지, 영상, 공간 데이터를 처리하는 데 특화
       - 주요 구성 요소:
       - 합성곱층(Convolution Layer): 커널(필터)로 특징 추출
       - 풀링층(Pooling Layer): 크기 축소, 연산량 감소
       - 완전연결층(FC Layer): 추출된 특징으로 최종 분류/예측 수행
       - 이미지 분류, 얼굴 인식, 자율주행, 의료 영상 분석 등
        

### (2) RNN (Recurrent Neural Network)
            
        - 순환 신경망
        - 시퀀스(순차적) 데이터 처리에 특화
        - 은닉 상태(Hidden State)를 통해 과거 정보를 현재 계산에 반영
        - 한계: 긴 시퀀스에서 과거 정보 손실 (→ LSTM, GRU로 보완)
        - 자연어 처리, 음성 인식, 시계열 예측 등

### 4️⃣ 공통 개념
         
       - 손실 함수(Loss Function): 예측과 실제 값 차이 측정
       - 역전파(Backpropagation): 오차를 거꾸로 전파하여 가중치 업데이트
       - 최적화 알고리즘(Optimizer): 가중치를 효율적으로 학습 (예: SGD, Adam)
       - 활성화 함수(Activation Function): 비선형성 추가 (예: ReLU, Sigmoid, Tanh)

## ❓ 어려웠던 부분  
채널과 차원의 차이점을 구분하는 데 조금 헷갈렸습니다. 

## 💬 질문 또는 토론 유도  
어려운 부분에서 연관된 질문입니다. 채널과 차원을 어떤 관점으로 보면 더 쉽게 이해할 수 있는 지 궁금합니다.

**Comment by parkjunohjjang12** [comment]
안녕하세요 문채린 님! 올려주신 내용 잘 읽었습니다.
CNN과 RNN의 핵심 포인트를 정확하게 잘 잡아주셨고, 특히 딥러닝과 전통 머신러닝의 차이를 명확히 구분해 주신 것 같습니다.
앞서 질문해 주신 어려웠던 부분에 대해서 피드백 드리도록 하겠습니다.

---

## 1️⃣ 어려웠던 부분 피드백

**채널과 차원의 차이점**

1. 채널

- 채널이란 데이터가 가진 특징(feature)의 개수를 의미합니다.  
- 가장 직관적인 예시는 이미지 데이터입니다. 우리가 보는 이미지는 픽셀(pixel) 단위로 이루어져 있고, 각 픽셀은 특정한 값을 가집니다.  
- 예를 들어, 이미지 크기가 32 × 32라고 하면 총 1,024개의 픽셀이 존재합니다. 이때 각 픽셀은 1개의 값만 가질 수도 있고, 여러 개의 값을 동시에 가질 수도 있습니다.  
- 이미지는 크게 흑백(grayscale)과 컬러로 나눌 수 있습니다.  
  - 흑백 이미지: 각 픽셀이 1개의 값(밝기 값)을 가짐 → 채널 수 = 1  
  - 컬러 이미지: 각 픽셀이 RGB(Red, Green, Blue)의 3가지 값으로 표현됨 → 채널 수 = 3  
- 즉, 픽셀이 몇 개의 값을 가지는지에 따라 채널 수가 달라진다고 이해하시면 됩니다.  
- 아래 그림처럼, R/G/B 각각의 채널 이미지를 합치면 우리가 보는 컬러 이미지가 완성됩니다.  


<img width="300" height="300" alt="Image" src="https://github.com/user-attachments/assets/2673a5f5-7f1e-4352-91b2-18f9c9edf63f" />

<img width="300" height="120" alt="Image" src="https://github.com/user-attachments/assets/05b85fab-5e18-494f-ad38-872cc244f5eb" />

2. 차원

 - 차원이란 데이터를 표현하기 위한 축(axis)의 개수를 의미합니다.  
- 예를 들어, 스칼라(Scalar)는 단일 값으로 차원이 없습니다. (0차원)  
- 벡터(Vector)는 값들이 일렬로 나열된 형태로, 보통 **1차원** 데이터라고 합니다.  
- 행렬(Matrix)은 행(row)과 열(column)로 이루어진 2차원 데이터입니다.  
- 텐서(Tensor)는 이 개념을 더 확장한 것으로, 3차원 이상을 표현할 수 있습니다.  
  - 예: 이미지 데이터는 (높이, 너비, 채널)의 3차원 구조를 가짐  
  - 동영상 데이터는 (시간, 높이, 너비, 채널)의 4차원 구조를 가짐

---

추가적으로 이해가 안 되시거나, 궁금한 내용들이 있으면 언제든지 편하게 질문주세요~!

감사합니다!! 😊
